### ISSUE TITLE
empty checkpoint folder are not removed

### ISSUE THREAD

**Issue by le1nux** [mlgym/mlgym#163]
When the DELETE DELETE /checkpoints/2023-04-05--11-42-41/0/4/optimizer is called and the folder is empty afterwards, it does not get removed. 

```
(base) root@:~/event_storage/2023-04-05--11-42-41/0# ls
0  1  2  3  4  experiment_config.json

root@:~/event_storage/2023-04-05--11-42-41/0# ls 0/

root@:~/event_storage/2023-04-05--11-42-41/0# ls 4/
lr_scheduler.pickle  model.pickle  optimizer.pickle  stateful_components.pickle

```

in the example above the folder 0 should be removed, as it is empty.

**Comment by moinam** [comment]
Hi Max, please correct me if I am wrong but my understanding was that this is an internal call made by mlgym, and when checkpoint_instruction.checkpoints_to_delete returns checkpoint ids, the files are deleted. The API calls are meant to delete the files and not the folders. 

Ex: If the user tries to delete optimizer, it will delete optimizer.pickle file. 

I will add the feature to check for empty folders after deleting pickel file, if folder is empty it will be deleted.

**Comment by le1nux** [comment]
yeah they are meant to delete the files. But if we only keep the checkpoint of the last epoch, we end up with hundreds of empty folders. That's why if the last pickle file in a folder is deleted, the folder should also be removed.
